packages/compass/src/app/index.tsx

```diff
@@ -221,13 +240,16 @@ const Application = View.extend({
       <React.StrictMode>
         <CompassElectron
           appName={remote.app.getName()}
-          getAutoConnectInfo={getAutoConnectInfo}
           showWelcomeModal={!wasNetworkOptInShown}
           createFileInputBackend={createElectronFileInputBackend(remote)}
-          onDisconnect={notifyMainProcessOfDisconnect}
+          onDisconnect={() => {
+            hasDisconnectedAtLeastOnce = true;
```

Comment:
Now that autoconnect method is not a hook dependency, do we still need this check? Is it because we have two hooks instantiated and both are calling the autoconnect?

Reply:
Actually here we're attempting to keep the current logic of auto-connect intact for Compass desktop which is to only auto-connect when it is first launch and we have not disconnected even once.

Earlier it was happening in Home itself with a React state `hasDisconnectedAtLeastOnce` but I moved it out of the Home assuming that this feature is very specific to Compass desktop and has nothing to do in general with Compass itself.

Reply:
Quick note on this as the discussion happened async on slack - We decided to remove the logic centred around the concept of wether the app was disconnected at-least once during the runtime and instead shifted to the idea that the `getAutoConnectInfo` should only be resolved once ever during the runtime of a renderer. This now happens in CompassRendererStorage using a small class property to track the same.